script/hooks/migration.js

```diff
@@ -1,167 +1,170 @@
+      } else if (!tokenData.actorLink && token.data.actorData.items) {
+        const update = migrateActorData(token.data.actorData, worldSchemaVersion);
+        console.log("ACTOR CHANGED", token.data.actorData, update);
+        tokenData.actorData = mergeObject(token.data.actorData, update);
+      }
+      return tokenData;
+    }),
+  };
 };
 
 export const migrateCompendium = async function (pack, worldSchemaVersion) {
-    const entity = pack.metadata.entity;
+  const entity = pack.metadata.entity;
 
-    await pack.migrate();
-    const content = await pack.getContent();
+  await pack.migrate();
+  const content = await pack.getContent();
 
-    for (let ent of content) {
-        let updateData = {};
-        if (entity === "Item") {
-            updateData = migrateItemData(ent.data, worldSchemaVersion);
-        } else if (entity === "Actor") {
-            updateData = migrateActorData(ent.data, worldSchemaVersion);
-        } else if (entity === "Scene") {
-            updateData = migrateSceneData(ent.data, worldSchemaVersion);
-        }
-        if (!isObjectEmpty(updateData)) {
-            expandObject(updateData);
-            updateData["_id"] = ent._id;
-            await pack.updateEntity(updateData);
-        }
+  for (let ent of content) {
+    let updateData = {};
+    if (entity === "Item") {
+      updateData = migrateItemData(ent.data, worldSchemaVersion);
+    } else if (entity === "Actor") {
+      updateData = migrateActorData(ent.data, worldSchemaVersion);
+    } else if (entity === "Scene") {
+      updateData = migrateSceneData(ent.data, worldSchemaVersion);
     }
+    if (!isObjectEmpty(updateData)) {
+      expandObject(updateData);
+      updateData["_id"] = ent._id;
+      await pack.updateEntity(updateData);
+    }
+  }
 };
 
 const migrateSettings = async function (worldSchemaVersion) {
-    if (worldSchemaVersion <= 3) {
-        game.settings.set("forbidden-lands", "showCraftingFields", true);
-        game.settings.set("forbidden-lands", "showCostField", true);
-        game.settings.set("forbidden-lands", "showSupplyField", true);
-        game.settings.set("forbidden-lands", "showEffectField", true);
-        game.settings.set("forbidden-lands", "showDescriptionField", true);
-        game.settings.set("forbidden-lands", "showDrawbackField", true);
-        game.settings.set("forbidden-lands", "showAppearanceField", true);
-    }
+  if (worldSchemaVersion <= 3) {
+    game.settings.set("forbidden-lands", "showCraftingFields", true);
+    game.settings.set("forbidden-lands", "showCostField", true);
+    game.settings.set("forbidden-lands", "showSupplyField", true);
+    game.settings.set("forbidden-lands", "showEffectField", true);
+    game.settings.set("forbidden-lands", "showDescriptionField", true);
+    game.settings.set("forbidden-lands", "showDrawbackField", true);
+    game.settings.set("forbidden-lands", "showAppearanceField", true);
+  }
+  if (worldSchemaVersion <= 3) {
```

Comment:
This needs to be changed to a correct version. Or removed if at all necessary. I have done tests without migrating the settings and haven't encountered any errors (the setting is still registered). 

Side Note: I noticed that the WorldSchemaVersion is a float and not int. Would we not be able to do decimal increments instead of whole number increments?